### PR TITLE
Eigenda/verifier

### PR DESCRIPTION
**Pull request by guoshijiang**

### CODE REVIEW

File: l2geth/rollup/sync_service.go [hunk 1]
```diff
@@ -1393,6 +1445,33 @@ func (s *SyncService) syncTransactionBatchRange(start, end uint64) error {
...
+		rollupInfo, err := s.eigenClient.GetRollupStoreByRollupBatchIndex(int64(i))
+		if err != nil {
+			return fmt.Errorf("cannot get rollup store by batch index: %w", err)
+		}
+		txs, err := s.eigenClient.GetBatchTransactionByDataStoreId(rollupInfo.DataStoreId)
+		if err != nil {
+			return fmt.Errorf("cannot get eigen transaction batch: %w", err)
+		}
+		for _, tx := range txs {
+			if tx.QueueOrigin().String() != "sequencer" || tx.GetMeta().Index == nil {
```
Comment (curryxbo) [inline l2geth/rollup/sync_service.go:1461]: Why skip L1ToL2 transactions? This type of transaction is found in Indexed Txs.
 If a transaction without Index is synchronized, the data source is not reliable,Whether to terminate the process？

File: l2geth/rollup/sync_service.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"github.com/mantlenetworkio/mantle/l2geth/rollup/eigenda"
```
Comment (curryxbo) [inline l2geth/rollup/sync_service.go:7]: pls format

File: mt-batcher/go.mod [hunk 3]
```diff
@@ -7,17 +7,25 @@ replace github.com/Layr-Labs/datalayr/common => ../datalayr-mantle/common
...
-	github.com/mantlenetworkio/mantle/bss-core v0.0.0-20221201061228-0589a659d047
 	github.com/mantlenetworkio/mantle/l2geth v0.0.0-20221201061228-0589a659d047
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.8.0
 	github.com/urfave/cli v1.22.10
 	google.golang.org/grpc v1.49.0
 )
 
+require (
+	// github.com/Layr-Labs/datalayr/common v0.0.0-00010101000000-000000000000
```
Comment (curryxbo) [inline mt-batcher/go.mod:20]: ditto

File: mt-batcher/metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"math/big"
```
Comment (curryxbo) [inline mt-batcher/metrics/metrics.go:7]: pls format

File: mt-batcher/mt_batcher.go [hunk 5]
```diff
@@ -2,22 +2,23 @@ package mt_batcher
...
-	"strings"
-
+	"github.com/mantlenetworkio/mantle/mt-batcher/metrics"
+	common2 "github.com/mantlenetworkio/mantle/mt-batcher/services/common"
+	"github.com/mantlenetworkio/mantle/mt-batcher/services/restorer"
+	"github.com/mantlenetworkio/mantle/mt-batcher/services/sequencer"
 	"github.com/urfave/cli"
 	"math/big"
-	"time"
+	"strings"
```
Comment (curryxbo) [inline mt-batcher/mt_batcher.go:21]: ditto

File: mt-batcher/mt_batcher.go [hunk 6]
```diff
@@ -26,103 +27,185 @@ func Main(gitVersion string) func(ctx *cli.Context) error {
...
+		ethc.Address(common.HexToAddress(cfg.EigenContractAddress)), parsed, l1Client, l1Client,
+		l1Client,
+	)
+	log.Info("contract init success", "EigenContractAddress", cfg.EigenContractAddress)
+
+	//mConfig := &metrics.MetricsConfig{
+	//	Hostname: cfg.MetricsHostname,
+	//	Port:     cfg.MetricsPort,
+	//}
+	//m := metrics.NewMetrics(mConfig)
```
Comment (curryxbo) [inline mt-batcher/mt_batcher.go:132]: pls clean up the useless code

File: mt-batcher/services/common/crypto.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
+package common
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"math/big"
+	"strings"
```
Comment (curryxbo) [inline mt-batcher/services/common/crypto.go:10]: pls format

File: mt-batcher/services/restorer/handle.go [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+	"bytes"
+	pb "github.com/Layr-Labs/datalayr/common/interfaces/interfaceRetrieverServer"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/labstack/echo/v4"
+	"github.com/mantlenetworkio/mantle/l2geth/core/types"
+	l2rlp "github.com/mantlenetworkio/mantle/l2geth/rlp"
+	"github.com/mantlenetworkio/mantle/mt-batcher/services/common"
+	"github.com/pkg/errors"
```
Comment (curryxbo) [inline mt-batcher/services/restorer/handle.go:13]: ditto

File: mt-batcher/services/restorer/service.go [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/mantlenetworkio/mantle/mt-batcher/bindings"
+	"github.com/mantlenetworkio/mantle/mt-batcher/services/common"
+	"strconv"
```
Comment (curryxbo) [inline mt-batcher/services/restorer/service.go:11]: pls format

File: mt-challenger/go.mod [hunk 10]
```diff
@@ -6,18 +6,22 @@ replace github.com/Layr-Labs/datalayr/common => ../datalayr-mantle/common
 
 replace github.com/Layr-Labs/datalayr/lib/merkzg => ../datalayr-mantle/lib/merkzg
 
+replace github.com/mantlenetworkio/mantle/mt-batcher => ../mt-batcher
+
 require (
-	github.com/Layr-Labs/datalayr/common v0.0.0-00010101000000-000000000000
+	// github.com/Layr-Labs/datalayr/common v0.0.0-00010101000000-000000000000
```
Comment (curryxbo) [inline mt-challenger/go.mod:12]: pls clean up the useless code

File: l2geth/rollup/sync_service.go [hunk 11]
```diff
@@ -776,6 +803,25 @@ func (s *SyncService) SetLatestBatchIndex(index *uint64) {
...
+	return *index + 1
+}
+
+func (s *SyncService) GetLatestEigenBatchIndex() *uint64 {
+	return rawdb.ReadLatestEigenBatchIndex(s.db)
+}
+
+func (s *SyncService) SetLatestEigenBatchIndex(index *uint64) {
+	if index != nil {
+		log.Info("set latest eigen batch index", "index", index)
```
[the remark this answers is not in the document]
Comment (guoshijiang) [reply l2geth/rollup/sync_service.go:820]: OK

File: l2geth/cmd/utils/flags.go [hunk 12]
```diff
@@ -831,7 +837,7 @@ var (
 	}
 	RollupBackendFlag = cli.StringFlag{
 		Name:   "rollup.backend",
-		Usage:  "Sync backend for verifiers (\"l1\" or \"l2\"), defaults to l1",
+		Usage:  "Sync backend for verifiers (\"l1\", \"l2\" or \"da\"), defaults to l1",
```
[the remark this answers is not in the document]
Comment (guoshijiang) [reply l2geth/cmd/utils/flags.go:840]: will remove in future

**Review by wukongcheng — APPROVED** [review]

**Review by curryxbo — APPROVED** [review]

**Review by bradyjoestar — APPROVED** [review]

**Pull request merged**